### ISSUE TITLE
[FR] Support for modeling non-standard residues

### ISSUE THREAD

**Issue by cxhernandez** [pandegroup/pdbfixer#91]
I don't know if this should be formally included in PDBFixer, but it might be nice to implement a plugin library (kinda like what wetmsm is to msmb) that allows PDBfixer to model non-standard residues.

I know for my own work, I've been interested in simulating acetylated/methylated residues and have had to rely on Rosetta for my pipeline. Rosetta works fine internally, but makes the code difficult to share and almost impossible to test on travis.

**Comment by cxhernandez** [comment]
Also, how difficult might this be to implement? It seems like most of the work come from curating more templates for PDBFixer to read.

**Comment by jchodera** [comment]
See also #44 and #31.

#44 has some useful concrete proposals.

Do you need to just model in these residues, or do you need to simulate
then in OpenMM afterwards too?

**Comment by cxhernandez** [comment]
Ah #44 seems ideal!

And yes, I would want to simulate the resulting structure in OpenMM.Hopefully, the pipeline would look similar to what we've already set up with [ParmEd and PyRosetta](http://parmed.github.io/ParmEd/html/omm_rosetta.html).

**Comment by cxhernandez** [comment]
Any progress on this (or #31/#44)? Would be great to have this working, as PyRosetta has become all but unusable at the moment.

**Comment by peastman** [comment]
I haven't worked on either of those.  If this is important for you, we can try to push up the priority of them.

**Comment by jchodera** [comment]
I think these would be great. Happy to contribute to their implementation, but we should be sure to define what the modes will do in a way we can all be happy with.

I still like the #44 proposals. What do you guys think?

**Comment by cxhernandez** [comment]
> If this is important for you, we can try to push up the priority of them.

Yes. This and *de novo* sequence generation (e.g. creating short linear peptides) would be very helpful.

**Comment by jchodera** [comment]
> Yes. This and de novo sequence generation (e.g. creating short linear peptides) would be very helpful.

Agreed, though you can easily do this right now with LEaP (which is installable via `conda` with `ambermini`).

**Comment by cxhernandez** [comment]
> Agreed, though you can easily do this right now with LEaP (which is installable via conda with ambermini).

Oh good point! Then #44 should have higher priority.
